builds/e2e/templates/e2e-run-metrics-collector.yaml

```diff
@@ -0,0 +1,64 @@
+steps:
+- pwsh: |
+    $testFile = '$(binDir)/Microsoft.Azure.Devices.Edge.Test.dll'
+
+    $filter = 'MetricsCollector'
+
+    if ($IsWindows)
+    {
+      # The windows installer script doesn't update the PATH in the current
+      # terminal session, so add `iotedge` and `docker` CLIs manually
+      $env:Path =
+        $env:Path,
+        $(Join-Path $env:ProgramFiles 'iotedge'),
+        $(Join-Path $env:ProgramFiles 'iotedge-moby') -join ';'
+      dotnet test $testFile --no-build --logger 'trx' --filter "$filter"
+    }
+    else
+    {
+      sudo --preserve-env dotnet test $testFile --no-build --logger 'trx' --filter "$filter"
+    }
+  displayName: Run tests
+  env:
+    E2E_DPS_GROUP_KEY: $(TestDpsGroupKeySymmetric)
+    E2E_EVENT_HUB_ENDPOINT: $(TestEventHubCompatibleEndpoint)
+    E2E_PREVIEW_EVENT_HUB_ENDPOINT: $(TestPreviewEventHubCompatibleEndpoint)
+    E2E_IOT_HUB_CONNECTION_STRING: $(TestIotHubConnectionString)
+    E2E_IOT_HUB_RESOURCE_ID: $(TestIotHubResourceId)
+    E2E_PREVIEW_IOT_HUB_CONNECTION_STRING: $(TestPreviewIotHubConnectionString)
+    E2E_REGISTRIES__0__PASSWORD: $(ReleaseContainerRegistryPassword)
+    E2E_ROOT_CA_PASSWORD: $(TestRootCaPassword)
+    E2E_BLOB_STORE_SAS: $(TestBlobStoreSas)
```

Comment:
If this YAML is redundant (and right now it is), it would be good to encapsulate it. However, (1) it might be better to let this new release process settle first before refactoring, and (2) many of these env vars aren't needed to run just the MetricsCollector test. If we strip it down to just the vars we need, then it is no longer redundant. Again, I'd wait for a bit, let things settle.